### ISSUE TITLE
[Feature] Pin files/folders

### ISSUE THREAD

**Issue by domanteli0** [cryptee/web-client#69]
**Is your feature request related to a problem? Please describe.**
Sometimes it takes time to find an important file inside a folder.

**Describe the solution you'd like**
Ability to pin things so that they would appear at the top of the list in the file explorer.

**Additional context**
I also just like having some files/folders being on top. For example.: a pined doc with my school schedule inside a folder for school stuff.

**Comment by johnozbay** [comment]
Hi there! ✌🏻
Thanks for filing this! 

If possible can you show me an example of this in use, _in a sortable view_, in any other file or document management system?

This idea came up a couple of times before, but it has its own set of issues, especially in the context of a sortable list-view, which is what Cryptee Docs has. 

Basically, the issues are : 

**How should the pinned/favorited files be sorted? Like all other files?**
So if things are sorted A-Z something like :
 
📌 Alligator Document
📌 Bird Document
📌 Cat Document
Alpaca Document
Bear Document
Camel Document

**What should happen if you're sorting by filetype?** 

Folders get sorted together, PDFs get sorted together etc. 
Should pinned files appear at the top of each filetype?
([Here's a screenshot of what I mean btw](https://github.com/cryptee/web-client/issues/37#issuecomment-588313667))
Like this:

📌 Alligator Document
Alpaca Document

📌 Bird.pdf
Bear.pdf

📌 Cat.mp3
Camel.mp3
etc.

**OR Should they be sorted on the top no matter what, because they're pinned, and ignore the sort anyway?**

📌 Alligator.pdf
📌 Bird.mp3
📌 Cat.zip

Alpaca.pdf

Bear.mp3

Camel.zip

If yes, wouldn't a folder labeled "All Pinned Docs 📌" (starting with "A", gets sorted first among folders, and has a 📌 in the folder name to make it visible), solve the problem? You can put all pinned files/documents into this, and quickly find them visually. 

Because otherwise – 

**What should happen if you have 100 pinned files, but 10 normal files?** 
Wouldn't that also make finding files difficult? 

If yes – 

**Should there be a limit for how many files you can pin?** 
Wouldn't this be an arbitrary limit and always dissatisfy some users?

Short version is that, there's no easy logical solution that works universally and fits everyone's needs here. Either we have to ignore the logical sort order of a folder – OR we have to put a number-limit on pinned files (to not have the same problem again). If we're ignoring the sort order, then accessing "pinned" files is as easy as creating a sub-folder.

What do you think?

**Comment by dhda** [comment]
My dream for a feature like this is to be able to have pinned shortcuts to documents that exist _outside of the folder hierarchy_. An example is that I keep a number of journal files to record thoughts and ideas that come up while working on other documents, and a “shortcut list” ~(possibly combined with a global “recent docs list”)~ that appears above the folder view and is visible regardless of your position in the folder hierarchy would be really helpful for easily and quickly switching back and forth between documents. On a desktop machine I use keyboard shortcuts + search to do this but on mobile I can’t do that, and backing all the way out of the current folder, then descending all the way into the folder for the other docs, and then going all the way back again, can get painful.

One of the really killer features of Cryptee for me right now is the ability to have a pdf document displayed over my Cryptee doc because it enables a similar kind of fast switching between documents that I haven’t really found in any other mobile apps. A pinned documents list ~(and/or a jump list in a more general form)~ would feel like a kindred productivity feature to that.

Edit: you know, I wrote this, and then went back to Cryptee, and realized that I had never actually clicked on the “Recent documents” tab in the sidebar so I feel pretty silly! I still think there’s utility in some kind of pinning feature, though I won’t feel as much need for it anymore.

**Comment by ghost** [comment]
I would also like to suggest a favorite file mark. Which means that some more important files can be pinned (or starred) to the main page, where the recent documents list is currently at.
